x-pack/plugins/rollup/public/crud_app/app.js

```diff
@@ -33,6 +33,11 @@ export class App extends Component {
     registerRouter(router);
   }
 
+  componentWillUnmount() {
+    // Set internal flag so we can prevent reacting to route changes internally.
+    setUserHasLeftApp(true);
```

Comment:
I had to implement this so that navigating to the fatal error screen didn't affect our routing logic.